ddcommon/src/container_id.rs

```diff
@@ -85,10 +89,27 @@ fn extract_container_id(filepath: &Path) -> Result<String, Box<dyn std::error::E
     Err(ContainerIdNotFoundError.into())
 }
 
+/// # Safety
+/// Must not be called in multi-threaded contexts
+pub unsafe fn set_cgroup_file(file: String) {
+    TESTING_CGROUP_PATH = Some(file)
+}
+
+fn get_cgroup_path() -> PathBuf {
+    // Safety: we assume set_cgroup_file is not called when it shouldn't
+    if let Some(path) = unsafe { TESTING_CGROUP_PATH.as_ref() } {
+        Path::new(path.as_str()).into()
+    } else {
+        Path::new(DEFAULT_CGROUP_PATH).into()
+    }
+}
+
 pub fn get_container_id() -> Option<&'static str> {
     // cache container id in a static to avoid recomputing it at each call
+
     lazy_static! {
-        static ref CONTAINER_ID: Option<String> = extract_container_id(Path::new(CGROUP_PATH)).ok();
+        static ref CONTAINER_ID: Option<String> =
+            extract_container_id(get_cgroup_path().as_path()).ok();
```

Comment:
Suggestion: Why not make the path an argument to this function? Callers that don't care about it can pass in `get_container_id(DEFAULT_CGROUP_PATH)`; callers that do can pass another one.

This way we eliminate the extra state, but allow the behavior to be configurable. If Rust had default values for arguments, this would be a textbook great use for them -- `get_container_id()` for the default and `get_container_id(custom)` for other use-cases, but as we say in Portugal, "if you don't have a hunting dog, use a cat" ;)

Reply:
Fair enough. (Sigh... Rust is so low-level lol, it's definitely not for me)